email_test.go

```diff
@@ -39,11 +41,100 @@ func TestRequireEnvConfig(t *testing.T) {
 		requiredVars[varName] = os.Getenv(varName)
 		os.Setenv(varName, "")
 	}
-	_, err := makeEmailConfigFromEnv()
+	_, err := makeEmailConfigFromEnv(api.Database)
 	if err == nil {
 		t.Errorf("should have received multiple error from unset env vars")
 	}
 	for varName, varValue := range requiredVars {
 		os.Setenv(varName, varValue)
 	}
 }
+
+func TestParseSESNotification(t *testing.T) {
+	inputs := [2]string{complaintJSON, bounceJSON}
+
+	for _, input := range inputs {
+		data := &blacklistRequest{}
+		err := json.Unmarshal([]byte(input), data)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if len(data.recipients) == 0 {
+			t.Error("failed to parse recipients from notification")
+		}
+		for _, recipient := range data.recipients {
+			if len(recipient.EmailAddress) == 0 {
+				t.Error("failed to parse email address from recipient")
+			}
+		}
+	}
+}
+
+func TestHandleSESNotification(t *testing.T) {
+	defer teardown()
+
+	resp, err := http.Post(server.URL+"/sns?amazon_authorize_key="+os.Getenv("AMAZON_AUTHORIZE_KEY"),
+		"application/json", strings.NewReader(complaintJSON))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		t.Errorf("SES notification without key should return 200, got %d", resp.StatusCode)
+	}
+
+	blacklisted, err := api.Database.IsBlacklistedEmail("complaint@simulator.amazonses.com")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !blacklisted {
+		t.Error("failed to blacklist email")
+	}
```

Comment:
could we add another sanity check to ensure `sendEmail` fails/short-circuits in the correct way?